pkg/api/validation/validation.go

```diff
@@ -373,9 +398,11 @@ func ValidateObjectMetaUpdate(newMeta, oldMeta *api.ObjectMeta, fldPath *field.P
 	allErrs = append(allErrs, ValidateImmutableField(newMeta.Namespace, oldMeta.Namespace, fldPath.Child("namespace"))...)
 	allErrs = append(allErrs, ValidateImmutableField(newMeta.UID, oldMeta.UID, fldPath.Child("uid"))...)
 	allErrs = append(allErrs, ValidateImmutableField(newMeta.CreationTimestamp, oldMeta.CreationTimestamp, fldPath.Child("creationTimestamp"))...)
+	allErrs = append(allErrs, ValidateImmutableField(newMeta.Finalizers, oldMeta.Finalizers, fldPath.Child("finalizers"))...)
```

Comment:
I think we have a fundamental disagreement about the power rating of update and delete. I think that update is strictly more powerful than delete, and therefore it's not surprising at all that update permission allows deletes; you guys seem to think the opposite? Is there prior art here? I'm not sure how you're going to convince me that power to update should not include power to delete.

Cascading deletions could be an issue only if the attacker also could add references willy-nilly. But that means the attacker already had control over those resources in the first place.